### ISSUE TITLE
[request] debounce test results or group before sending

### ISSUE THREAD

**Issue by ranisalt** [travis-ci/travis-ci#5288]
Yesterday I accidentally ran `git push --all` forgetting that I had 15 branches that I haven't pushed to github yet on my local repo. Shortly after, I started receiving emails from Travis as if I was plagued by spam.

![Travis](http://i.imgur.com/ndHE3ty.png)
Those emails came within 5 minutes and became kind of annoying. I suggest the emails should be [debounced](https://en.wiktionary.org/wiki/debounce), that is, Travis wait for about 5 minutes and group all logs to send them together instead of sending all of them at once.

There may be better ways to solve that using different debouncing algorithms or throttling, but it surely would prevent that.

Thanks!

**Comment by BanzaiMan** [comment]
I'm sorry that you got a lot of notifications, but you did request them (unintentionally). But imagine you did this intentionally and didn't get notifications for some of them. You might equally be upset. Sadly, we cannot tell the difference.

**Comment by ranisalt** [comment]
@BanzaiMan yes, I do acknowledge I want those notifications, but I suggest sending them all at once. If Travis is expecting results from another job, then I think it's ok to postpone and send them all at once. Mind that I'm not saying to not send them, just to send all at once.

**Comment by BanzaiMan** [comment]
Do you have `fast_finsih: true`? If so, then this is a duplicate of #4928.
